Apps/WebClient/src/NewClientApp/src/stores/vaccinationStatusAuthenticated.ts

```diff
@@ -0,0 +1,387 @@
+                    });
+            }
+        }
+
+        function retrieveAuthenticatedVaccineRecord(
+            hdid: string
+        ): Promise<void> {
+            const logger = container.get<ILogger>(SERVICE_IDENTIFIER.Logger);
+            const vaccinationStatusService =
+                container.get<IVaccinationStatusService>(
+                    SERVICE_IDENTIFIER.VaccinationStatusService
+                );
+
+            logger.debug(`Retrieving authenticated vaccination record`);
+            setAuthenticatedVaccineRecordRequested(hdid);
+
+            return vaccinationStatusService
+                .getAuthenticatedVaccineRecord(hdid)
+                .then((result) => {
+                    const payload = result.resourcePayload;
+                    if (result.resultStatus === ResultType.Success) {
+                        logger.info("Authenticated vaccination record loaded");
+                        setAuthenticatedVaccinationRecord(hdid, payload);
+                    } else if (
+                        result.resultError?.actionCode === ActionType.Refresh &&
+                        !payload.loaded &&
+                        payload.retryin > 0
+                    ) {
+                        logger.info(
+                            "Authenticated vaccination record not loaded but will try again"
+                        );
+                        setAuthenticatedVaccineRecordStatusMessage(
+                            hdid,
+                            "Please wait a moment while we download your proof of vaccination."
+                        );
+                        setTimeout(() => {
+                            logger.info(
+                                "Re-querying for downloading the authenticated vaccination record"
+                            );
+                            retrieveAuthenticatedVaccineRecord(hdid);
+                        }, payload.retryin);
+                    } else {
+                        if (result.resultError) {
+                            throw result.resultError;
+                        }
+                        logger.warn(
+                            `Authenticated vaccination record retrieval failed! ${JSON.stringify(
+                                result
+                            )}`
+                        );
+                    }
+                })
+                .catch((error: ResultError) => {
+                    logger.error(
+                        "Authenticated vaccination record not loaded due to unexpected error"
+                    );
+                    handleAuthenticatedPdfError(
+                        hdid,
+                        error,
+                        ErrorType.Retrieve
+                    );
+                });
+        }
+
+        function authenticatedVaccineRecordState(
+            hdid: string
+        ): VaccineRecordState {
+            return getAuthenticatedVaccineRecordState(hdid);
+        }
```

Comment:
I think this should be grouped with the getters.

Reply:
Grouped under computed.